meta-mender-core/classes/mender-helpers.bbclass

```diff
@@ -350,19 +357,36 @@ def get_extra_parts_by_id(d, id):
 
 def get_extra_parts_labels(d):
     labels = []
-    parts = get_extra_parts(d)
-    if parts:
-        for part in parts:
-            s = part.find("--label")
-            if s > 0:
-                e = part.find("--", s + 2)
-                labels.append(part[s + len("--label="):e])
+    ids = get_extra_parts_flags(d)
+    if ids:
+        for id in ids:
+            labels.append(get_extra_parts_label(d, id))
     return ' '.join(labels)
 
+def get_extra_parts_label(d, id = None):
+    part = get_extra_parts_by_id(d, id)
+    if part:
+        s = part.find("--label")
+        if s > 0:
+            e = part.find("--", s + 2)
+            return part[s + len("--label="):e]
+    return ""
+
 def get_extra_parts_wks(d):
     final_parts = []
     parts = get_extra_parts(d) or {}
     if parts:
         for part in parts:
             final_parts.append("part --ondisk \"$ondisk_dev\" --align \"$alignment_kb\" {}".format(part))
     return '\n'.join(final_parts)
+
+def get_extra_parts_fstab(d):
+    out = []
+    extra_parts_offset = mender_get_extra_parts_offset(d)
+    device = d.getVar('MENDER_STORAGE_DEVICE_BASE')
+    for part in get_extra_parts_flags(d):
+       label = get_extra_parts_label(d, part)
+       out.append("{:20s} {:20s} {:10s} {:21s} 0 0".format(device + str(extra_parts_offset), "/mnt/{}".format(label), "auto", "default"))
```

Comment:
@kacf this is also not necessary to align as same is done in fstab postprocess callback and I tested it and work fine. Will drop this in final version something like:
```
--- a/meta-mender-core/classes/mender-helpers.bbclass
+++ b/meta-mender-core/classes/mender-helpers.bbclass
@@ -399,7 +399,7 @@ def get_extra_parts_fstab(d):
     for part in get_extra_parts_flags(d):
        label = get_extra_parts_label(d, part)
        fstype_opts = get_extra_parts_fstab_opts(d, part).split()
-       out.append("{:20s} {:20s} {:10s} {:21s} 0 0".format(device + str(extra_parts_offset), "/mnt/{}".format(label), fstype_opts[0], fstype_opts[1]))
+       out.append("{} {} {} {} 0 0".format(device + str(extra_parts_offset), "/mnt/{}".format(label), fstype_opts[0], fstype_opts[1]))
        extra_parts_offset += 1

```

Reply:
Ok!